### ISSUE TITLE
Func<Task> as method param gives compiler error

### ISSUE THREAD

**Issue by persn** [zompinc/sync-method-generator#66]
Consider the following code

```csharp
[Zomp.SyncMethodGenerator.CreateSyncVersion]
public async Task FooAsync(Func<int, Task> bar)
{
    await bar(5);
}
```

It will generate the following
```csharp
public void Foo(global::System.Func<int> bar)
{
    bar(5);
}
```

This will fail to compile because the `int` has suddenly become a return value instead of a param for the delegate `error CS1593: Delegate 'Func<int>' does not take 1 arguments`

A Func<..., Task> should probably be transformed to a `Action` when `Task` is the return value.

**Comment by virzak** [comment]
Will be available in 1.3.36 in the next 10 minutes.

This case was already being handled in the local variable declaration, so the same logic was reused for parameter.

**Comment by virzak** [comment]
Thanks for submitting
